mitmproxy/net/dns/https_records.py

```diff
@@ -63,30 +65,6 @@ def _unpack_params(data: bytes, offset: int) -> dict[int, bytes]:
     return params
 
 
-def _unpack_target_name(data: bytes, offset: int) -> tuple[str, int]:
-    """Unpacks the DNS-encoded domain name from data starting at the given offset."""
-    labels = []
-    while True:
-        if offset >= len(data):
-            raise struct.error("unpack requires a buffer of %i bytes" % offset)
```

Comment:
Creating this PR because this line had no dedicated test and `individual coverage` passed because the fuzz tests covered this line. Figured I could use the existing API itself instead of having to write new tests (should have done this in the initial PR though)